### PR TITLE
CLDSRV-452: Id and principal on policy hotfixbranch

### PR DESCRIPTION
**Pull request by benzekrimaha**
# Pull request template

### Motivation and context

This PR adds the arsenal version that support of the Id definition on resource policy document and principal length restriction on both resource and user policy

Ticket link : https://scality.atlassian.net/browse/CLDSRV-452
ARSN PR link : https://github.com/scality/Arsenal/pull/2175

**Due to failing CI , fixups were made on tests.yaml and s3cmd tests.js**

Tests were added as well

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -204,54 +204,35 @@ jobs:
...
     runs-on: ubuntu-latest
     needs: build
     env:
       S3BACKEND: file
       S3VAULT: mem
       CLOUDSERVER_IMAGE: ghcr.io/${{ github.repository }}/cloudserver:${{ github.sha }}
       MPU_TESTING: "yes"
-      ENABLE_NULL_VERSION_COMPAT_MODE: "${{ matrix.enable-null-compat }}"
-      JOB_NAME: ${{ matrix.job-name }}
+      #ENABLE_NULL_VERSION_COMPAT_MODE: "${{ matrix.enable-null-compat }}"
```
Comment (ghost) [inline .github/workflows/tests.yaml]: Depending on how atomic the commits that introduced these change son the usual branches are, maybe cherry-picking them will be easier for maintenance next time :slightly_smiling_face:

File: tests/functional/aws-node-sdk/test/bucket/putBucketPolicy.js [hunk 2]
```diff
@@ -44,6 +71,7 @@ function assertError(err, expectedErr, cb) {
     cb();
 }
 
+
```
Comment (ghost) [inline tests/functional/aws-node-sdk/test/bucket/putBucketPolicy.js]:
```suggestion
```

**Review by ghost — APPROVED** [review]

**Review by ghost — APPROVED** [review]

File: .github/workflows/tests.yaml (hunk 1 above)
Comment (nicolas2bert) [inline .github/workflows/tests.yaml]: Let's keep running the tests for both `ENABLE_NULL_VERSION_COMPAT_MODE` values.

Comment (benzekrimaha) [reply .github/workflows/tests.yaml]: Like discussed, the tests were put back

**Review by nicolas2bert — APPROVED** [review]

**Comment by benzekrimaha** [comment]
@bert-e approve

**Pull request merged**